listener/upload_test.go

```diff
@@ -74,3 +75,32 @@ func TestUpdateSystemPlatform(t *testing.T) {
 
 	deleteData(t)
 }
+
+func TestParseUploadMessage(t *testing.T) {
+	msg := createTestingUploadKafkaMsg()
+	event, identity, err := parseUploadMessage(msg)
+	assert.Nil(t, err)
+	assert.Equal(t, "TEST-00000", event.Id)
+	assert.Equal(t, "User", identity.Identity.Type)
+}
+
+func TestUploadHandler(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+	configure()
+	deleteData(t)
+
+	msg := createTestingUploadKafkaMsg()
+	uploadHandler(msg)
+
+	var system models.SystemPlatform
+	assert.Nil(t, database.Db.Where("inventory_id = ?", "TEST-00000").Find(&system).Error)
```

Comment:
Could you please add more assertions, for user account, and for different attributes of SystemPlatform like last_seen being close to now, and last_evaluated being null ?